### DISCUSSION TITLE
camu.select / select issue

### DISCUSSION THREAD

**Discussion by christophertrapani** [Ircam-RnD/catart-mubu#88] [Bug, Issues, and Suggestions]
Hi all — as you might've seen in my long message on the IRCAM Forum I'm delving into selection.

I'm noticing the following with mubu.knn and wonder if it is a bug or intended.

new patch loaded, 3 columns sent, followed by select message with 3 values then "gettarget": first time, the results look as intended

<img width="347" alt="Screenshot 2023-05-04 at 2 47 45 PM" src="https://user-images.githubusercontent.com/35881355/236209651-788e6f04-33f9-43b1-8706-3236bb2f9550.png">

second time, three new column values are sent, but camu.select (or mubu.knn?) retains the initial three values.

<img width="633" alt="Screenshot 2023-05-04 at 2 48 00 PM" src="https://user-images.githubusercontent.com/35881355/236210217-3563c5bb-10ae-4d20-8262-00042ed83e57.png">

Is this expected behavior? Is there a clear command I should insert? Patch below; Thanks

———
<pre><code>
----------begin_max5_patcher----------
468.3ocyUssahCCD84juhQ9sUJMJ14Rg9qTUsxDrnoahcjuzkUU8eu1iCpkR
nPEUnUHFzLdBGeN9LNujlPVo1JLD3N3dHI4kzjDrTnPxTdBYfussmav1HCBi
guQPxhqYEasX8BXuO6VejaaerSt42ZQqMhSSdQFzPCwE0gHikW.OL8.R2Pmr
WXQzXSE6ViXnV8zMzZx6cpb1csRmpFKY+2nHhFg.ODV40zzPH6LoYqZXPHsG
PSV9u9BpUtXIRHLThwhiwM5LbqbdtU7CR.5WRfkW19m8ys++jOat8ZUdcFPK
qhdnlusUp4p3jN1.Sqp2MHMPMz.2lAFQumYPQNq1GBeoYvFO9bsOdFdtnLTw
91xvs+WHCTfAkWfLr7BUAVY0UQGjh+5g6.YXvsxk+GoD7q2pziNCrVXZ0mjw
wKRo0m3lz4lVmmuUGiu8cF+AvQ9cOs.2B95xO+pEjEg56KPFkS2tClIKI7NQ
7BgsSxscJ4G5gF6Y1ifyEG7P+pfT30VWGf7+Gkm.G+Ee94sCOw3iiOKzlolQ
H7l2mT5P5hLLsSFSqwTs34tc8iFGBW68oVuI0owsEYaSbvhLnVKzRWGZGSCj
yCINXH49aGF4QdfyOoul9FfV90Oj
-----------end_max5_patcher-----------
</code></pre>

**Comment by diemoschwarz** [comment]
yes, works as intended.  Only the chosen columns have non zero weights.

**Comment by christophertrapani** [comment]
But in a patch where columnas and weights are being set independently, if I
change weights to > 0., I can have a "phantom" target value that I cannot
control or systematically clear, which (when often changing weights and
columns) can be problematic.
